src/services/coderefactorings/inlineTempRefactor.ts

```diff
@@ -0,0 +1,158 @@
+/* @internal */
+namespace ts.coderefactoring {
+    registerCodeRefactoring({
+        getCodeActions: (context: CodeRefactoringContext): CodeAction[] => {
+            const sourceFile = context.sourceFile;
+            const identifier = getTokenAtPosition(sourceFile, context.span.start);
+            let variableDeclaration: VariableDeclaration;
+            if (identifier.kind !== SyntaxKind.Identifier || !(isVariableDeclaration(identifier.parent))) {
+                // this refectoring only works on a variable declarations
+                return undefined;
+            }
+            else {
+                variableDeclaration = identifier.parent;
+            }
+
+            const fileTextChanges: FileTextChanges[] = [];
+            const namePos: number = variableDeclaration.name.pos;
+            let variableInitializerText: string = variableDeclaration.initializer.getText();

```

Comment:
This is not going to work. the text from the initializer is not guaranteed to work in all places where the variable would be used. the initializer expression can contain names that have been shadowed later on and will need to be qualified accordingly . consider:

```ts
namespace N {
     export var x = 0;

     var temp = x;

    function inner(x: string) {
        console.log(temp);
    }
}
```

renaming `temp` to `x` is not correct, since `x` has a different meaning in this scope.